source/margarita/upgrading.rst

```diff
@@ -0,0 +1,557 @@
+   we can easily copy files we need:
+
+   .. code-block:: bash
+
+      ~/dev$ git clone git@github.com:caktus/django-project-template.git
+      ~/dev$ ls
+      django-project-template myproject
+      ~/dev$ cd myproject
+      ~/dev/myproject$
+
+
+#. Copy some important files and directory to our project:
+
+   .. code-block:: bash
+
+      ~/dev/myproject$ mv fabfile.py fabfile.py.old
+      ~/dev/myproject$ cp ../django-project-template/fabfile.py .
+      ~/dev/myproject$ cp ../django-project-template/Vagrantfile .
+      ~/dev/myproject$ cp ../django-project-template/install_salt.sh .
+      ~/dev/myproject$ cp ../django-project-template/conf/master.tmpl conf/
+      ~/dev/myproject$ rm conf/master.conf
+      ~/dev/myproject$ cp ../django-project-template/conf/gpg.tmpl conf/
+
+#. Edit fabfile.py:
+
+   1. Make sure ``SALT_VERSION`` is ``2015.5.8``
+   2. Change ``env.master`` in both the ``staging()`` and ``production()`` functions to be
+      correct (get the value from fabfile.py.old)
+
+
+Upgrade Margarita
+-----------------
+
+#. Add to ``conf/pillar/project.sls``:
+
+   .. code-block:: yaml
+
+      margarita_version: 1.5.0
+      less_version: 1.5.1   # or whatever you're currently using
+      postgres_version: 9.1 # or whatever you're currently using
+
+#. Move the old states out of the way (don't leave them in the conf directory), and copy in the
+   minimal new list:
+
+   .. code-block:: bash
+
+      ~/dev/myproject$ mv conf/salt salt.old
+      ~/dev/myproject$ cp -r ../django-project-template/conf/salt conf/
+
+
+Single Deploy settings
+----------------------
+
+Our current deployment expects all Django settings to be in a single module named ``deploy.py``.
+We need to merge the ``staging.py`` and ``production.py`` files into one called ``deploy.py``.
+The easiest way is to create a new file called ``deploy.py`` with this content:
+
+.. code-block:: python
+
+   import os
+   ENVIRONMENT = os.environ['ENVIRONMENT']
+   if ENVIRONMENT == 'staging':
+       from .staging import *
+   elif ENVIRONMENT == 'production':
+       from .production import *
+   else:
+       from .local import *
+
+That should be refactored ASAP to get rid of the staging and production files.
+
+Dotenv
+------
+
+#. Add ``myproject/load_env.py`` (same dir as root ``urls.py``):
+
+   .. code-block:: python
+
+      from os.path import dirname, join
+      import dotenv
+
+
+      def load_env():
+          "Get the path to the .env file and load it."
+          project_dir = dirname(dirname(__file__))
+          dotenv.read_dotenv(join(project_dir, '.env'))
+
+#. Modify ``myproject/celery.py``:
+
+   .. code-block:: diff
+
+      Modified   myproject/celery.py
+      diff --git a/myproject/celery.py b/myproject/celery.py
+      index d9a4e87..4f5a199 100644
+      --- a/myproject/celery.py
+      +++ b/myproject/celery.py
+      @@ -11,2 +11,5 @@ from celery import Celery
+
+      +from . import load_env
+      +load_env.load_env()
```

Comment:
I guess not, this is how we have it in RescueID: https://github.com/theirc/rescue_id/blob/develop/rescue_id/celery.py#L5-L6

Should I add a line?